fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -1177,6 +1177,29 @@ public void testNoUnion() {
         }
     }
 
+    @Test
+    public void testDryRunSaveRecord() throws Exception {
+        CompletableFuture<FDBStoredRecord<Message>> future1;
+        TestRecords1Proto.MySimpleRecord.Builder recBuilder = TestRecords1Proto.MySimpleRecord.newBuilder();
+        TestRecords1Proto.MySimpleRecord record1 = recBuilder.setRecNo(1).build();
+        TestRecords1Proto.MySimpleRecord record2 = recBuilder.setRecNo(2).setNumValue3Indexed(3).build();
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+            future1 = recordStore.dryRunSaveRecordAsync(record1, FDBRecordStoreBase.RecordExistenceCheck.ERROR_IF_EXISTS);
+            // assert returns the message to be saved
+            assertEquals(record1, future1.get().getRecord());
```

Comment:
If we care enough about the `SizeInfo` to calculate it correctly, it feels like we should also assert on it. We could, for example, take the `SizeInfo` from an `FDBStoredRecord` that we "saved" with a dry update, and then compare it to the `SizeInfo` of an actual save of the same record that we do later.